api/v1/vouchers/models.py

```diff
@@ -0,0 +1,59 @@
+import random
+from datetime import datetime
+from flask.ext.sqlalchemy import SQLAlchemy
+
+from stdnum import verhoeff
+
+from ..exceptions import VoucherException
+db = SQLAlchemy()
+
+__author__ = 'kenneth'
+
+
+class Voucher(db.Model):
+    __tablename__ = 'voucher_vouchers'
+
+    id = db.Column(db.Integer, primary_key=True)
+    code = db.Column(db.String(6))
+    redeemed_on = db.Column(db.DateTime(timezone=True), nullable=True)
+    created_on = db.Column(db.DateTime(timezone=True), server_default=db.func.now())
+    modified_on = db.Column(db.DateTime(timezone=True), server_default=db.func.now(), server_onupdate=db.func.now())
+    redeemed_by = db.Column(db.String(13), nullable=True)
+
+    def __init__(self, code):
+        self.code = code
+
+    def __repr__(self):
+        return self.code
+
+    @classmethod
+    def create(cls):
+        voucher = cls(code=cls.generate_code())
+        db.session.add(voucher)
+        db.session.commit()
+        return voucher
+
+    @classmethod
+    def redeem(cls, code, phone):
+        voucher = cls.query.filter_by(code=str(code)).first()
+        if voucher is None:
+            raise VoucherException("Voucher does not exist")
+        if voucher.redeemed_on is not None:
+            raise VoucherException("Attempting to redeem an already redeemed voucher")
+        voucher.redeemed_on = datetime.now()
```

Comment:
It will always default to UTC, which I think is fine and much simpler than trying to find what timezone the server is sitting on. Especially considering that we will never display this on the web